### PR TITLE
Add auto heading id filter to the wysiwyg format on install.

### PR DESCRIPTION
**Pull request by rupertj**
Fix for #142

### CODE REVIEW

File: localgov_publications.install [hunk 1]
```diff
@@ -5,17 +5,35 @@
...
   $path_auto_config->set('safe_tokens', $safe_tokens)->save();
+
+  // Add heading_id_filter from the auto_heading_ids module to the wysiwyg
+  // filter format. This is required for the ToC block to work consistently.
+  /** @var \Drupal\filter\Entity\FilterFormat $wysiwygFormat */
+  $wysiwygFormat = FilterFormat::load('wysiwyg');
+  $wysiwygFormat->setFilterConfig('heading_id_filter', [
+    'status' => TRUE,
+  ]);
+  $wysiwygFormat->save();
```
Comment (andybroomfield) [inline localgov_publications.install]: Just to flag that this format might not exist if a site builder has deleted it becuase they use a different format. So check first that it exists before trying to update it.

Comment (finnlewis) [reply localgov_publications.install]: Roger that!

File: localgov_publications.install [hunk 2]
```diff
@@ -5,17 +5,37 @@
...
   $path_auto_config = \Drupal::configFactory()->getEditable('pathauto.settings');
   $safe_tokens = $path_auto_config->get('safe_tokens');
   $safe_tokens[] = 'localgov-publication-cover-page-alias';
   $path_auto_config->set('safe_tokens', $safe_tokens)->save();
+
+  // Add heading_id_filter from the auto_heading_ids module to the wysiwyg
+  // filter format. This is required for the ToC block to work consistently.
+  /** @var \Drupal\filter\Entity\FilterFormat $wysiwygFormat */
+  $wysiwygFormat = FilterFormat::load('wysiwyg');
+  if ($wysiwygFormat) {
```
Comment (andybroomfield) [inline localgov_publications.install]: `if ($wysiwygFormat instanceof FilterFormatInterface)` and add at the top `use Drupal\filter\FilterFormatInterface`

**Comment by rupertj** [comment]
I've made an issue and a patch to add the functionality we need to auto_heading_ids here: https://www.drupal.org/project/auto_heading_ids/issues/3419508

File: localgov_publications.install (hunk 2 above)
Comment (rupertj) [reply localgov_publications.install]: Good plan. Done.

**Comment by rupertj** [comment]
I've copied in the patched version of the filter we're using from auto_heading_ids and removed it as a dependency. If they accept the patch, I'll consider taking our copy out and reverting back to using auto_heading_ids.

**Comment by finnlewis** [comment]
I'll test this!

**Review by finnlewis — APPROVED** [review]
Works a treat, thanks @rupertj !

**Comment by rupertj** [comment]
Excellent. Thanks @finnlewis!

**Pull request merged**